### ISSUE TITLE
May i get some help before starting porting NuttX to a new arch namely LoongArch

### ISSUE THREAD

**Issue by GooTal** [apache/nuttx#7566]
Hello there,
I\`m trying to port NuttX to LoongArch, which is used in 64-bit microcontroller.
Currently i`m a student and hope to learn some OS knowledge during porting. 

Is it a good choice to take riscv-k210 as a referrence?
I\`m planning to read documents from official NuttX website. 
I\`ve glanced some code snippets and found the code for uart, interrupt and so on. But i didn`t find out how the memoy page table was established (Sorry for my poor knowledge). Can i get some help about memory mapping? Like: the layout of virtual memory.

Could you tell me something about the proper process of porting NuttX?  Maybe start with head.S, uart,  memory mapping...and something else?

Thanks a lot :)

**Comment by acassis** [comment]
Hi @GooTal I think LoongArch (from 龙芯) is based on MIPS instruction set. We have support for MIPS, but not for MIPS 64-bit yet. Since riscv-k210 is RISC-V 64-bit and RISC-V ISA is similar to MIPS maybe it could be a good starting point.

You can use our old Porting Guide ( https://cwiki.apache.org/confluence/display/NUTTX/Porting+Guide ) as reference, but always comparing with the source code because some functions name and files could have changed or moved to other places. If you can document the Porting Process to create a new documentation to include on our new Documentation site: https://nuttx.apache.org/docs/latest/ it will be welcome!

**Comment by GooTal** [comment]
Alright, thank you!

**Comment by GooTal** [comment]
Hi, i want to make sure that NuttX is fully open source, right?
I investigated another RTOS namely rt-thread, but some libraries of rt-thread is prebuilt (not open source), thus making it difficult for me to debug or continue learning.
Thanks for your time again.

**Comment by GooTal** [comment]
Hi @acassis , is the [porting guide](https://cwiki.apache.org/confluence/display/NUTTX/Porting+Guide) still in maintenance? 

Is there a git repo for the porting guide? Maybe it\`s more convenient to maintain and discuss details of the doc in a git repo.

Thanks.

**Comment by xiaoxiang781216** [comment]
> Hi @acassis , is the [porting guide](https://cwiki.apache.org/confluence/display/NUTTX/Porting+Guide) still in maintenance?
> 

Here is he new location: https://nuttx.apache.org/docs/latest/reference/os/index.html

> Is there a git repo for the porting guide? Maybe it`s more convenient to maintain and discuss details of the doc in a git repo.
> 

Yes, it's here: https://github.com/apache/nuttx/tree/master/Documentation
you can send the change to github directly, the website will automatically update to contain your modification once the patch get merged.

> Thanks.

**Comment by xiaoxiang781216** [comment]
> Hi, i want to make sure that NuttX is fully open source, right?

Yes, all OS component and the bundle apps(https://github.com/apache/nuttx-apps) are open source with Apache license(some are MIT/BSD etc), you can reference here for more info:
https://github.com/apache/nuttx/blob/master/LICENSE

> I investigated another RTOS namely rt-thread, but some libraries of rt-thread is prebuilt (not open source), thus making it difficult for me to debug or continue learning. Thanks for your time again.

But some chip drivers(especially WiFi/BLE firmware) is released by the static library.

**Comment by GooTal** [comment]
Thanks a lot @xiaoxiang781216!

**Comment by GooTal** [comment]
Hi, i\`m trying to add some config items in Kconfig files. I notice that there is a `$ARCH` variable in `${TOPDIR}/Kconfig` file. I think it may stand for host cpu arch. But i couldn\`t find out where the definition is. 

Here are the steps:

In ${TOPDIR}
```
./tools/configure.sh -l maix-bit:nsh
make menuconfig
```
As you can see, the mainmenu is "NuttX/x86_64 Configuration" 
![图片](https://user-images.githubusercontent.com/40800103/210514835-143c9153-4a73-4e08-8d6a-801cdbf4e855.png)


Thanks for your time~

**Comment by TimJTi** [comment]
I believe this is set via the “System Type” menu at the top level.

 

You may know this (I only recently learnt about it) but if not then: if you type a forward slash while in the menuconfig it brings up a search. If you search for ARCH it points you to line 161 of arch/Kconfig which is always one way of finding out what’s available and where. Each search result also has a numbered result – if you type in that number it takes you to the actual menuconfig menu for that result.

**Comment by GooTal** [comment]
Hey, i noticed that the aclint base for risc-v qemu is defined like this:
```
#define QEMU_RV_ACLINT_BASE  0x02f00000
```
I can`t find out the related specification, but only found this https://github.com/riscv/riscv-aclint/releases/download/v1.0-draft2/riscv-aclint-1.0-draft2.pdf. It seems this doc does not mentioned the aclint base addr 0x02f00000.

Can you provide some info about the related specification?

**Comment by xunmi-git** [comment]
Hi @GooTal 

I'm Xunmi from the LoongArch Community. We noticed your previous work on porting NuttX to LoongArch and would love to know:

Is this effort still active?
Could you share the current status/progress?
If helpful, our community can provide hardware resources (e.g., LoongArch-based testing devices) to support this initiative. Looking forward to hearing from you!
